### ISSUE TITLE
Rendering with pointclouds obtained from image

### ISSUE THREAD

**Issue by SwagJ** [facebookresearch/pytorch3d#528]
## ❓ Questions on how to use PyTorch3D

Hi,

**TASK**:
perform 3d transformation on pointcloud generated from image and render transformed pcd and project it back as an image 

**Problem**
a. rendered image is incorrect for this following process:
        1. generate coordinate list of pixels in real world coordinate system, using intrinsic and extrinsic. load corresponding RGB 
             values. 
        2. init everything follows this tutorial: https://pytorch3d.org/tutorials/render_colored_points
    Then, the output image is totally black. 

b. is that possible to make output image height and width as different value?

**Code**:
`import os
import sys
import torch
import torch.nn.functional as F
import matplotlib.pyplot as plt
from skimage.io import imread
from PIL import Image

import numpy as np

from pytorch3d.structures import Pointclouds
from pytorch3d.vis.plotly_vis import AxisArgs, plot_batch_individually, plot_scene
from pytorch3d.renderer import (
    look_at_view_transform,
    FoVOrthographicCameras, 
    PointsRasterizationSettings,
    PointsRenderer,
    PulsarPointsRenderer,
    PointsRasterizer,
    AlphaCompositor,
    NormWeightedCompositor
)

def pixel_coord_np(width, height):
    x = np.linspace(0, width - 1, width).astype(np.int)
    y = np.linspace(0, height - 1, height).astype(np.int)
    [x, y] = np.meshgrid(x, y)
    return np.vstack((x.flatten(), y.flatten(), np.ones_like(x.flatten())))

def depth_read(filename):
    depth_png = np.array(Image.open(filename), dtype=int)
    assert(np.max(depth_png) > 255)

    depth = depth_png.astype(np.float) / 256.
    depth[depth_png == 0] = -1.
    return depth

if torch.cuda.is_available():
    device = torch.device("cuda:0")
    torch.cuda.set_device(device)
else:
    device = torch.device("cpu")

color_raw = imread('./kitti/rgb_1.png')
depth = depth_read('./kitti/completed_depth_kitti.png')
intrinsics_file = read_raw_calib_file('./kitti/calib_cam_to_cam.txt')
intrinsics_raw = np.reshape(intrinsics_file['P_rect_02'],(3,4))
intrinsics = intrinsics_raw[:3,:3]
inv_intrinsic = np.linalg.inv(intrinsics)

intrinsics_full = np.vstack((intrinsics_raw,[0,0,0,1]))

R = intrinsics_file['R_rect_03'].reshape(3,3)
T = intrinsics_file['T_02']

cam_matrix = intrinsics @ np.hstack((R,T.reshape(3,1)))
print(cam_matrix)
inv_cam_matrix = np.linalg.inv(np.vstack((cam_matrix,[0,0,0,1])))
print(inv_cam_matrix)

height = depth.shape[0]
width = depth.shape[1]

image_coord = pixel_coord_np(width,height)
cam_coords = (inv_intrinsic[:3, :3] @ image_coord * depth.flatten())
cam_coords_homo = np.vstack((cam_coords,np.ones((1,cam_coords.shape[1]))))
world_coords_homo = inv_cam_matrix @ cam_coords_homo
world_coords = world_coords_homo.T[:,:3]
print("world_coords shape:",world_coords.shape)
rgb = color_raw.reshape(height*width,3)

cam_coords_tensor = torch.Tensor(cam_coords.T).to(device)
world_coords_tensor = torch.Tensor(world_coords).to(device)
rgb_tensor = torch.Tensor(rgb).to(device)/255.
T_tensor = torch.Tensor(T).unsqueeze(0)
R_tensor = torch.Tensor(R).unsqueeze(0)
K_tensor = torch.Tensor(intrinsics_full).unsqueeze(0)
print("cam_coords_tensor shape:",cam_coords_tensor.shape)
point_cloud = Pointclouds(points=[world_coords_tensor], features=[rgb_tensor])

cameras = FoVOrthographicCameras(device=device, R=R_tensor, T=T_tensor, K=K_tensor)
raster_settings = PointsRasterizationSettings(
    image_size=height, 
    radius = 0.003,
    points_per_pixel = 1
)

rasterizer = PointsRasterizer(cameras=cameras, raster_settings=raster_settings)

renderer = PointsRenderer(
    rasterizer=rasterizer,
    compositor=AlphaCompositor()
)

images = renderer(point_cloud)
print(images.shape)

plt.figure(figsize=(10, 10))
plt.imshow(images[0, ..., :3].cpu().numpy())
plt.grid("off")
plt.axis("off")`

Theoretically, it should be recovering original image. Did I use a wrong render? 
For your convenience, I have attack the input image, depth map & cam_calib. 

I am looking forward to your reply. Thank you very much!

Best,
[calib_cam_to_cam.txt](https://github.com/facebookresearch/pytorch3d/files/5844971/calib_cam_to_cam.txt)
![completed_depth_kitti](https://user-images.githubusercontent.com/22762687/105227885-22945080-5b62-11eb-99a4-b84588f58057.png)
![rgb_1](https://user-images.githubusercontent.com/22762687/105227889-232ce700-5b62-11eb-9c56-e0d4f9fa2d5c.png)

**Comment by nikhilaravi** [comment]
@SwagJ 

Yes you can set the image/height and width to be different values. In the `PointsRasterizationSettings` you should specify a tuple of (height, width). 

The other thing to check is whether the camera coordinate conventions of your data match that of PyTorch3D - here is a useful doc with diagrams to explain this more: https://github.com/facebookresearch/pytorch3d/blob/master/docs/notes/cameras.md

**Comment by abhigoku10** [comment]
@SwagJ  did u try visualizing the lidar point cloud data ie bin files using pytorch3D

**Comment by gkioxari** [comment]
No response. Closing this issue.
